crates/solver/src/solver/oneinch_solver.rs

```diff
@@ -74,16 +80,50 @@ impl OneInchSolver {
             allowance_fetcher: Box::new(AllowanceManager::new(web3, settlement_address)),
             protocol_cache: ProtocolCache::default(),
             oneinch_slippage_bps,
+            max_slippage_in_wei,
         })
     }
 }
 
 impl OneInchSolver {
-    /// Settles a single sell order against a 1Inch swap using the specified protocols.
-    async fn settle_order_with_protocols(
+    /// Computes the max slippage we are willing to use for a given trade to limit the absolute
+    /// slippage to a configured upper limit in terms of wei. Because 1Inch keeps positive
+    /// slippage, always applying a default slippage would otherwise become very costly for huge
+    /// orders.
+    fn compute_max_slippage(
+        external_buy_token_price_in_wei: &BigRational,
+        buy_amount: &U256,
+        default_slippage_bps: u32,
+        max_slippage_in_wei: &U256,
+    ) -> Result<Slippage> {
+        let max_slippage_in_buy_token =
+            max_slippage_in_wei.to_big_rational() / external_buy_token_price_in_wei;
+
+        let max_slippage_percent_respecting_wei_limit =
+            max_slippage_in_buy_token / buy_amount.to_big_rational();
+
+        let max_slippage_bps_respecting_wei_limit =
+            max_slippage_percent_respecting_wei_limit * BigRational::from_u128(10_000).unwrap();
+
+        let final_slippage_bps = std::cmp::min(
+            max_slippage_bps_respecting_wei_limit
+                .to_u32()
```

Comment:
Will this give zero if `max_slippage_bps_respecting_wei_limit` is smaller than 1? This should be fine, and I guess it would just result in setting the final slippage to zero, right?

Reply:
Yes and yes.
However, we could also round the `max_slippage_bps_respecting_wei_limit` instead of truncating it to the nearest integer. Not sure what is more desirable, though. What do you think?